src/object/obj_enum.c

```diff
@@ -729,8 +724,7 @@ clear_iod_csum(struct dcs_iod_csums *iod_csum)
 		return;
 
 	for (i = 0; i < iod_csum->ic_nr; i++)
-		if (iod_csum->ic_data[i].cs_csum != NULL)
-			D_FREE(iod_csum->ic_data->cs_csum);
+		D_FREE(iod_csum->ic_data->cs_csum);
```

Comment:
This code looks wrong.

Reply:
Agree. It looks this should be "D_FREE(iod_csum->ic_data[i].cs_csum)".